src/vs/workbench/contrib/scm/browser/scmViewService.ts

```diff
@@ -108,32 +116,51 @@ export class SCMViewService implements ISCMViewService {
 			}, 0)
 	);
 
-	private _focusedRepository: ISCMRepository | undefined;
-
 	get focusedRepository(): ISCMRepository | undefined {
-		return this._focusedRepository;
+		return this._repositories.find(r => r.focused)?.repository;
 	}
 
 	private _onDidFocusRepository = new Emitter<ISCMRepository | undefined>();
 	readonly onDidFocusRepository = this._onDidFocusRepository.event;
 
-	private _compareRepositories: (op1: ISCMRepository, op2: ISCMRepository) => number;
+	private _repositoriesSortKey: ISCMRepositoryViewSortKey;
+	private _compareRepositories: (op1: ISCMRepositoryView, op2: ISCMRepositoryView) => number;
 
 	constructor(
-		@ISCMService private readonly scmService: ISCMService,
+		@ISCMService scmService: ISCMService,
 		@IInstantiationService instantiationService: IInstantiationService,
 		@IStorageService private readonly storageService: IStorageService,
-		@IWorkspaceContextService workspaceContextService: IWorkspaceContextService
+		@IWorkspaceContextService workspaceContextService: IWorkspaceContextService,
+		@IConfigurationService configurationService: IConfigurationService,
 	) {
 		this.menus = instantiationService.createInstance(SCMMenus);
 
-		this._compareRepositories = (op1: ISCMRepository, op2: ISCMRepository): number => {
-			const name1 = getRepositoryName(workspaceContextService, op1);
-			const name2 = getRepositoryName(workspaceContextService, op2);
+		try {
+			this.previousState = JSON.parse(storageService.get('scm:view:visibleRepositories', StorageScope.WORKSPACE, ''));
+		} catch {
+			// noop
+		}
+
+		this._repositoriesSortKey = this.previousState?.sortKey ?? configurationService.getValue('scm.repositories.sortOrder');
```

Comment:
Should we do some validation here? The configuration service will gladly return whatever the user sets in settings, which can be any value. Validation would be to make sure it's always within the `'discovery time' | 'name' | 'path'` set.

Reply:
Fixed. In case of an invalid value, we will now default to `discovery time`.